### PR TITLE
chore(site): replace xstate by react-query on update password

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

File: site/src/pages/UserSettingsPage/SecurityPage/SecurityPage.tsx [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
-  const [securityState, securitySend] = useMachine(
-    userSecuritySettingsMachine,
-    {
-      context: {
-        userId: me.id,
-      },
-    },
-  );
-  const { error } = securityState.context;
+  const updatePasswordMutation = useMutation(updatePassword());
```
Comment (Parkreiner) [inline site/src/pages/UserSettingsPage/SecurityPage/SecurityPage.tsx:18]: Does this need to invalidate any queries in the query client, or is the rest of the querying logic set up to update things frequently enough that it's not much of an issue?

Comment (BrunoQuaresma) [reply site/src/pages/UserSettingsPage/SecurityPage/SecurityPage.tsx:18]: No we don't because we don't store the password anywhere.

Comment (Parkreiner) [reply site/src/pages/UserSettingsPage/SecurityPage/SecurityPage.tsx:18]: Okay, cool. That was my only concern. I'm going to approve this.

**Review by buenos-nachos — APPROVED** [review]

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/9598

**Pull request merged**